its/core-it-support/core-it-plugins/maven-it-plugin-active-collection/src/main/java/org/apache/maven/plugin/coreit/CheckThreadSafetyMojo.java

```diff
@@ -95,7 +95,10 @@ public void run() {
                     getLog().info("[MAVEN-CORE-IT-LOG] Thread " + this + " uses " + tccl);
                     Thread.currentThread().setContextClassLoader(tccl);
                     while (go.isEmpty()) {
-                        // wait for start
+                        try {
+                            Thread.sleep(100); // wait for the start
+                        } catch (InterruptedException ignored) {
```

Comment:
Empty blocks are illegal wrt checkstyle.

Reply:
yes this rules must be absent or bug. As we have this several times i assume its misconfigured.

- https://checkstyle.sourceforge.io/checks/blocks/emptyblock.html
- https://github.com/apache/maven-shared-resources/blob/bad8df9df24313a66dd5968194b631305ac499dd/src/main/resources/config/maven_checks_nocodestyle.xml#L106
- so its a bug. Again one reason to chain the tools.

Reply:
> Empty blocks are illegal wrt checkstyle.

if this is true then this whole PR would not be possible as the root cause is https://pmd.github.io/pmd/pmd_rules_java_codestyle.html#emptycontrolstatement.

![Uploading image.png…]()


[INFO] BUILD SUCCESS